### PR TITLE
Package/py-lmfit: add new version

### PR DESCRIPTION
**Pull request by Sinan81**
add new versions, and add new dependencies

Dependency `py-asteval` will be created soon.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-lmfit/package.py [hunk 1]
```diff
@@ -12,8 +12,15 @@ class PyLmfit(PythonPackage):
...
-    depends_on('py-numpy@1.5:',  type=('build', 'run'))
-    depends_on('py-scipy@0.14:', type=('build', 'run'))
-    depends_on('py-setuptools',  type='build')
+    depends_on('py-asteval@0.9.16',         type=('build', 'run'), when='@0.9.15:')
+    depends_on('py-numpy@1.5:',             type=('build', 'run'), when='@0.9.5:')
+    depends_on('py-numpy@1.16:',            type=('build', 'run'), when='@0.9.15:')
+    depends_on('py-scipy@0.14:',            type=('build', 'run'), when='@0.9.5')
+    depends_on('py-scipy@1.2:',             type=('build', 'run'), when='@0.9.15:')
+    depends_on('py-setuptools',             type='build')
+    depends_on('py-six@1.11:',              type=('build', 'run'), when='@0.9.15: ^python@2.7:2.8')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lmfit/package.py]:
```suggestion
    depends_on('py-six@1.11:',              type=('build', 'run'), when='@0.9.15')
```
It isn't needed for 1.0.1, and it is needed for 0.9.15 when Python 3.

File: var/spack/repos/builtin/packages/py-lmfit/package.py [hunk 2]
```diff
@@ -12,8 +12,15 @@ class PyLmfit(PythonPackage):
...
+    depends_on('py-uncertainties@3.0.1:',   type=('build', 'run'), when='@0.9.15:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lmfit/package.py:28]: Can you add:
```python
depends_on('python@3.5:', type=('build', 'run'), when='@1:')
depends_on('python@2.7:2.8,3.5:', type=('build', 'run'), when='@0.9.15')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
py-uncertainties doesn’t seem to have a Spack package yet

**Comment by adamjstewart** [comment]
Can you rebase now that `py-uncertainties` has been merged?

**Comment by Sinan81** [comment]
@adamjstewart Do you know why the  macos test is failing? The CI details didn't make sense to me.

**Comment by adamjstewart** [comment]
That’s a known bug we’re working on. You can ignore it.

**Pull request merged**